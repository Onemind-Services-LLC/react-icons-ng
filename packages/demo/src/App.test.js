import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";

import App from "./App";

it("renders without crashing", () => {
  render(<App />);
});

it("snapshot test", () => {
  const { container } = render(<App />);
  expect(container.firstChild).toMatchSnapshot();
});

it("always includes a title", () => {
  const { container } = render(<App />);
  const icons = container.getElementsByTagName("svg");
  for (let i = 0; i < icons.length; i++) {
    expect(icons[i].getElementsByTagName("title").length).toBe(1);
  }
});
