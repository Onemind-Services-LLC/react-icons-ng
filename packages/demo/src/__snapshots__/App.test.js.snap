// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  class="App"
>
  <header
    class="App-header"
  >
    <img
      alt="logo"
      class="App-logo"
      src="logo.svg"
    />
    <h1
      class="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <p
    class="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
    <svg
      class="global-class-name"
      fill="currentColor"
      focusable="false"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: blue;"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Icon
      </title>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m9 7h-6v13h-2v-6h-2v6H9V9H3V7h18z"
      />
    </svg>
    <svg
      class="global-class-name additional-class-name"
      color="red"
      fill="currentColor"
      focusable="false"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: red;"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        folder icon
      </title>
      <path
        d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Icon
      </title>
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M12 2c1.1 0 2 .9 2 2s-.9 2-2 2-2-.9-2-2 .9-2 2-2m9 7h-6v13h-2v-6h-2v6H9V9H3V7h18z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Icon
      </title>
      <path
        d="M464 128H272l-64-64H48C21.49 64 0 85.49 0 112v288c0 26.51 21.49 48 48 48h416c26.51 0 48-21.49 48-48V176c0-26.51-21.49-48-48-48z"
      />
    </svg>
    <svg
      baseProfile="tiny"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      version="1.2"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        arrow down icon
      </title>
      <path
        d="M16.707 13.293c-.391-.391-1.023-.391-1.414 0l-2.293 2.293v-7.586c0-.552-.447-1-1-1s-1 .448-1 1v7.586l-2.293-2.293c-.391-.391-1.023-.391-1.414 0s-.391 1.023 0 1.414l4.707 4.707 4.707-4.707c.391-.391.391-1.023 0-1.414z"
      />
    </svg>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Icon
      </title>
      <path
        d="M12,2 C6.485,2 2,6.485 2,12 C2,17.515 6.485,22 12,22 C17.515,22 22,17.515 22,12 C22,6.485 17.515,2 12,2 Z"
        fill="none"
        stroke="#865CD6"
        stroke-width="4"
      />
    </svg>
  </p>
</div>
`;
